cmd/execution_builder.go

```diff
@@ -1424,8 +1425,8 @@ func getContractEpochCounter(
 		return 0, fmt.Errorf("could not read epoch counter, script returned no value")
 	}
 
-	epochCounter := output.Value.ToGoValue().(uint64)
-	return epochCounter, nil
+	epochCounter := output.Value.(cadence.UInt64)
```

Comment:
just curious why is this better? prevents mistyping the cadence type?

Reply:
See https://github.com/onflow/cadence/pull/3291, i.e. https://github.com/onflow/cadence/pull/2531#issuecomment-2075423041: 

There is no "right" Go equivalent for all Cadence values. Developers disagree on what `ToGoValue` should return, but code (like here) essentially relies on a certain behaviour, and the return type is simply `any`. This essentially makes code using the function very fragile. 

As there cannot be a "right" `Value.ToGoValue` function for all values, and to prevent this fragility, we removed the `ToGoValue` helper function. Code using `cadence.Value` knows best what Go type they would like the Cadence value be converted to.